### DISCUSSION TITLE
Problem with passing var to templets

### DISCUSSION THREAD

**Discussion by PiotrCembrowski** [a-h/templ#353] [Q&A]
```go
u := model.User{
    Email: "losttape@gmail.com",
}
mux.Handle("/dashboard/user", templ.Handler(user.User(u)))
```

i got error cannot convert u (variable of type model.User) to type user.User

user.User is just a templ component that looks like this:

```go
package user

import (
    "github.com/piotrcembrowski/go-api/model"
    "github.com/piotrcembrowski/go-api/view/dashboard"
)

templ User(u model.User) {
    @dashboard.Dashboard() {
        <h1>Hello {u.Email} from User page </h1>
    }
}
```

any ideas what went wrong?

**Comment by a-h** [comment]
It's hard to say what went wrong without seeing more of the code. In the first part of the code, it could be that the imports are wrong, or it could be that you forgot to run `templ generate` again after updating the `u` parameter of the `User` templ component to be a `model.User`.

Without the context, it's impossible to see.

**Comment by a-h** [comment]
Closed, since no further information was provided.
